docs/cross-platform.md

```diff
@@ -0,0 +1,44 @@
+# Cross-Platform Development with `esy`
+
+## Goal
+
+The goal for `esy` is to work equally well on Linux, Windows, and OSX - without needing special cases or hacks.
+
+The development workflow for `esy` should be the same for all of these platforms.
+
+## Architecture
+
+One challenge with the OCaml ecosystem is that many of the scripts rely heavily on 'bashisms', like `make`, bash shell scripts, etc.
+
+Unfortunately, those tools aren't available out-of-the-box on Windows. WSL is starting to offer more capabilities around this,
+but there isn't a seamless install story. An important piece of meeting our goal stated above is that the installation process
+for `esy` is simple - `npm install -g esy`.
+
+To that end, we use `cygwin` on Windows to provide a transparent layer for running these 'bashisms'. You don't even have to know about it -
+we install it silently, keep it in a sandbox, and delegate commands to it under the hood- but you don't need to touch it for `esy`. As we
+improve native package support, we might not even need this layer long-term, but it helps get us started with a seamless windows experience.
+
+### Path considerations
+
+A major difference with Windows vs OSX/Linux is the path separator - `\` vs `/`. Windows can actually work with _either_ path separator (for the most part). 
+
+The challenge, though, is `\` is also the __escape character__ - so we can hit double-escaping issues when we render the strings.
+
+For the time being, we'll __normalize all paths to use a forward slash__ - this keeps us from hitting the 'double-escaping issues' with backslash.
+
+One other complexity is that we use Cygwin for running our build tasks, and the canonical paths for Cygwin are POSIX style, like:
+- `C:\temp` -> ``/cygdrive/c/temp`
+
+However, the majority of Cygwin utilities can work with either style of paths - therefore, we will use the the Windows path, normalized with a forward slash, for shell invocations.
+
+For utilities that do not support this - like `rsync` - we will convert the path to a cygwin-style path. Cygwin includes the `cygpath` utility for this.
```

Comment:
Is this line correct? We convert to a cygwin style path because rsync does *not* support forward slashes? I don't quite understand.

Reply:
Ah yes, this wasn't very clear. `rsync` doesn't understand any windows path - whether it's a forward or backslash - so `C:/temp` and `C:\\temp` won't work, but `/cygdrive/c/temp` will.